### PR TITLE
Adopt new paths filter and refactor CI

### PR DESCRIPTION
**Pull request by Superskyyy**

### CODE REVIEW

File: .github/workflows/publish-docker.yaml
Comment (kezhenxu94) [inline .github/workflows/publish-docker.yaml:1]: We still need this workflow, did you forget to add it in the all-in-one workflow file?

Comment (Superskyyy) [reply .github/workflows/publish-docker.yaml:1]: My bad, an accident.

**Review by kezhenxu94 — APPROVED** [review]

File: .github/workflows/publish-docker.yaml
Comment (kezhenxu94) [reply .github/workflows/publish-docker.yaml:1]: The whole file is changed, can you check whether you changed the line breaks or not? For example did you change from \n to \r\n or so

**Comment by kezhenxu94** [comment]
Please check the publish-docker.yaml file

**Comment by Superskyyy** [comment]
> Please check the publish-docker.yaml file

It's probably due to I deleted it first then added it back, then git thinks the files, although exactly the same, are two different entities.

**Review by kezhenxu94 — APPROVED** [review]
Never mind, LGTM

**Pull request merged**